lib/tasks/send_coupons.rake

```diff
@@ -0,0 +1,9 @@
+desc "it sends coupon code to all users"
+task send_coupons: :environment do
+  User.find_each do |user|
+    if coupon = Coupon.active.find_by(user: user)
```

Comment:
create an association/method on user named active_coupons and use that here

Reply:
Yes sir, fixing.